### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by AyomipoMajek**
OOP school library: set up associations done according to microverse requirements which include:

Create a class Classroom with the following:
@label instance variable, should be initialized in the constructor.
Setter and getter for @label (remember about attr_accessor).
Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
Create the has-many side (a classroom has many students).
Create the belongs-to side (a student belongs to a classroom).
Make sure that when adding a student to a classroom it also sets the classroom for the student.
Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
Create a class Book with the following:
@title and @author instance variables, should be initialized in the constructor.
Setters and getters for instance variables (remember about attr_accessor).
Create a class Rental with the following:
@date instance variable, should be initialized in the constructor.
Setter and getter for @date (remember about attr_accessor).
Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
Create the has-many side of Book and Rental (a book has many rentals).
Create the belongs-to side of Rental and Book (a rental belongs to a book).
Create the has-many side of Person and Rental (a person has many rentals).
Create the belongs-to side of Rental and Person (a rental belongs to a person).
Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

**Review by victorokoroji — CHANGES_REQUESTED** [review]
Hi @AyomipoMajek , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## ❌ Invalid Code Review Request

You have submitted a project that does not have a professional readme. This is so because you included the live demo section in your readme and you are not deploying the project. Kindly remove the live demo section and submit again 👍🏽 


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  end
+end
+
+class Student < Person
+  attr_accessor :classroom
+
+  def initialize(age, parent_permission: true, name: 'Unknown')
+    super(age, parent_permission: parent_permission, name: name)
+    @classroom = nil
+  end
```
Comment (1GORDON) [inline classroom.rb:21]: Great job @AyomipoMajek 🤝

- [ ] When setting the classroom for a student it also adds it to the classroom's students. Kindly add a `method that serves as the 'belongs-to' side (a student belongs to a classroom)`. Check out this [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships) for more guidance.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Book
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+  end
+end
```
Comment (1GORDON) [inline book.rb]: Great job @AyomipoMajek 🤝

- [ ] Kindly create the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. Forexample a `book has many rentals`.
- [ ] Kindly ensure a method for adding rentals should be present. Follow this [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) for more guidance.
- [ ] Also ensure the rentals variable is initialsed. You may also follow this [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/class_and_object_for_irb.md#constructor-and-instance-vars) for ore guidance.

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
+
+class Person
+  attr_accessor :name, :age, :rentals
+
```
Comment (1GORDON) [inline rental.rb:25]: Great job @AyomipoMajek 🤝

- [ ] Kindly ensure there is a method for adding rentals. Kindly follow this [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) for more guidance.

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @AyomipoMajek ✋

Good job so far!

## Highlights!
- Followed Gitflow ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️

- [x] I would suggest each individual Class is implemented in its own `.rb` file. This will help with the organization of your project.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classroom.rb (hunk 1 above)
Comment (AyomipoMajek) [reply classroom.rb:21]: Done, thank you

File: book.rb (hunk 2 above)
Comment (AyomipoMajek) [reply book.rb]: Done, thanks

File: rental.rb (hunk 3 above)
Comment (AyomipoMajek) [reply rental.rb:25]: Done, thanks

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @AyomipoMajek 

Your project is complete! Great job for making the changes🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- associations well created
- No linter errors

----[optional] For a cleaner and clear code it will be better if you create all your classes in their separate files.

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

tufoin

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**